### PR TITLE
[6.1.0]Let `aquery` print effective environment for all `CommandAction`s

### PR DESCRIPTION
**Pull request by kshyanashree**
Instead of printing the fixed part of the `ActionEnvironment` of all `SpawnActions`, `aquery` now prints the effective environment resolved against an empty client environment of all `AbstractAction`s that implement `CommandAction`.

For `SpawnAction`s, this should not result in any observable change, but C++ actions, which only override `AbstractAction#getEffectiveEnvironment`, now have their fixed environments (e.g. toolchain env vars such as `INCLUDE` with MSVC) emitted.

Work towards #12852

Closes #17108.

PiperOrigin-RevId: 501198850
Change-Id: I035a8cfde32193ca7f1f71030bd183c20edfdc0d

### CODE REVIEW

**Review by meisterT — APPROVED** [review]

**Comment by meisterT** [comment]
Related tests are failing. Did you investigate why?

**Review by meisterT — CHANGES_REQUESTED** [review]
(removing approval until we have figured out why tests are failing)

File: src/test/shell/integration/aquery_test.sh [hunk 1]
```diff
@@ -1659,6 +1665,49 @@ EOF
   fi
 }
 
+function test_does_not_fail_horribly_with_file() {
```
Comment (fmeum) [inline src/test/shell/integration/aquery_test.sh:1668]: @kshyanashree This test made it in by mistake, only the one after it is part of the original PR.

Comment (kshyanashree) [reply src/test/shell/integration/aquery_test.sh]: Hi @fmeum! I removed this test and ran the presubmits, but still checks are failing. Could you please look into it?

**Comment by keertk** [comment]
@meisterT checks have passed - could you please (re)approve?

**Review by meteorcloudy — APPROVED** [review]

**Review by meisterT — APPROVED** [review]

**Pull request merged**